File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> m;

        for(int i=0; i<n; i++){
            m[nums[i]] = i;
        }

        for(int i=0; i<n; i++){
            int diff = target-nums[i];

            if(m.find(diff) != m.end()){
                if(m[diff] != i){
                    return {i, m[diff]};
                }
            }
        }

        return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        lo, hi = 0, n-1

        while lo <= hi:
            mid = lo + (hi-lo)//2

            if nums[mid] == target:
                return mid

            if nums[mid] > target:
                hi = mid - 1
            else:
                lo = mid + 1


        return -1